httpcore/_exceptions.py

```diff
@@ -19,6 +19,10 @@ class ConnectionNotAvailable(Exception):
     pass
 
 
+class ServerDisconnectedError(Exception):
```

Comment:
It could be subclass of `ConnectionNotAvailable` since there connection won't be available after server closed it. this would prevent from [this change](https://github.com/encode/httpcore/pull/928/files#diff-2c49ff24c06fb1f90e1cba7d18757d80f3197a92106e04ba77179497461d69aa).

Also, we need to add it in docs.

Reply:
Done, made it a subclass of the existing one. Should we have a subclass also for the previous state related error?

> Also, we need to add it in docs.

I can not see existing one mentioned in the docs. Probably due to the exception being handled by the pool. I added docstrings instead to the classes about this.

Reply:
Im now wondering do we even need the new `ServerDisconnectedError`. Because this is also internal to the pooling mechanism like the previous one.